add support for remote repos

Rizziepit opened the pull request.

Rizziepit commented:
@miltontony ready for review

(Tests are failing because new elastic-git hasn't been deployed by Travis yet)

miltontony commented on development.ini:
```diff
@@ -15,8 +15,15 @@ pyramid.includes =
+# The index on Elasticsearch corresponds to the basename (minus .git or .json suffix).
+# For that reason, the basename must be alphanumeric and lowercase. It may also not
+# start with _.
+git.path = %(here)s/repo
```
Can we get rid of this and use `git.content_repo_url` in the same way that springboard does?

Rizziepit replied:
I tried that, but `git.path` and `git.content_repo_url` are used by eg-tools commands. So I thought it best to leave them as they are so that we still have all those commands at our disposal.

miltontony replied:
@Rizziepit but it looks like `content_repo_url` is not used anywhere?

Rizziepit replied:
You're right. It was used by `init_repository` but that has been deleted. Want me to remove it? As long as we keep `git.path` we'll have backwards compatibility.

Rizziepit commented:
@miltontony ping on this PR. It's passing now

Rizziepit commented:
@miltontony ready for review again. I removed `content_repo_url`, added the `pull` task that replaces `fastforward` and moved `/api/notify/` + tests to core views. The entire api module has been deleted.

miltontony commented:
\o/ yay for deleteing code and :+1:
